### ISSUE TITLE
Feature Request: Cut & paste blocks

### ISSUE THREAD

**Issue by boehsermoe** [luyadev/luya-module-cms#160]
When I want to move a block from inside a container block, which is at the bottom of the page, to the top of the page (or to another container block). Currently I have to add the block to the clipboard and move it to the new placeholder and than remove the old. 
Here would be cut & paste function useful.

**Comment by nadar** [comment]
Cut & Paste sounds interesting to me, the cut button would be easy done - but where and how does the Paste button look like?

**Comment by boehsermoe** [comment]
After click the "cut" button the "paste mode" will be active and could look like this:
<img width="638" alt="screenshot 2018-11-16 at 18 54 12" src="https://user-images.githubusercontent.com/2466744/48638499-1ab05300-e9d1-11e8-8a45-97181b92ecc0.png">

Either the "paste area" could be visibly instantly for all blocks or only on hover one block. After paste or canceled the "paste mode" will disabled.

**Comment by nadar** [comment]
Yes, this could work, i was thinking about the same solution - maybe it could look a bit differente, i will draw something.

**Comment by TheMaaarc** [comment]
I think we should go with a button shape for the "paste here" action, because you need to click and not drag & drop the element, right?

**Comment by boehsermoe** [comment]
@TheMaaarc Yes right, I agree.

**Comment by nadar** [comment]
@boehsermoe cut + paste is equal to "moving the block by dragging the block the the top" for large pages, yes this is sometimes a little heavy task as you have to click and scroll at the same time. 

Instead of adding another button, we could just make the page scroll up and down when we are close the edge of the screen. So we could improve the drag&drop behavior instead of providing two solutions for the same task, and also not clutter the UI.

**Comment by boehsermoe** [comment]
Maybe it is be enough if the page will scroll up and down while dragging. We should try this at first it will improve the UI anyway. 

But I think it would be much better UX if the block can be move/cutting by a context menu like this:
![uifinal](https://user-images.githubusercontent.com/2466744/53621649-7eb0fd00-3bf7-11e9-8ff6-ce746e8c18da.gif)

Other idea is that there are two kind of modes. One for editing the block content and add blocks etc. and a second for extended layout editing. In layout mode the block preview is only the name or id of the block to make it easier too drag&drop the block.

**Comment by TheMaaarc** [comment]
Shouldn't be too difficult to integrate a context menu – just needs time.  
But would be a nice addition to hide stuff like copy, hide and possibly future functionality.
